site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java

```
@@ -10724,32 +10724,20 @@ private void saveFeatures(ParameterParser params, SessionState state, Site site)
 						addAlert(state, rb.getString("java.addalias"));
 					} else {
 						try {
-							// first, clear any alias set to this channel
-							AliasService.removeTargetAliases(channelReference); // check
-							// to
-							// see
-							// whether
-							// the
-							// alias
-							// has
-							// been
-							// used
+							String target = AliasService.getTarget(alias);
+							boolean targetsThisSite = site.getReference().equals(target) ||
+									channelReference.equals(target);
+							if (!(targetsThisSite)) {
+								addAlert(state, rb.getString("java.emailinuse") + " ");
+							}
+						} catch (IdUnusedException ee) {
 							try {
-								String target = AliasService.getTarget(alias);
-								boolean targetsThisSite = site.getReference().equals(target);
-								if (!(targetsThisSite)) {
-									addAlert(state, rb.getString("java.emailinuse") + " ");
-								}
-							} catch (IdUnusedException ee) {
-								try {
-									AliasService.setAlias(alias,
-											channelReference);
-								} catch (IdUsedException exception) {
-								} catch (IdInvalidException exception) {
-								} catch (PermissionException exception) {
-								}
+								AliasService.setAlias(alias,
+										channelReference);
+							} catch (IdUsedException exception) {
+							} catch (IdInvalidException exception) {
```

Can you log these exceptions instead of swallowing the exceptions?